HW2/P2/P2.txt

```diff
@@ -0,0 +1,14 @@
+As the P2.png plot shows, the serial code is clearly faster than the multi-threaded code in this instance, regardless
+ of whether the numbers are correlated or not. There is not much difference between the computation time of the
+ serial code in the uncorrelated versus correlated case.
+
+ The fine-grained locking benefited slightly (but noticeably) from having the numbers correlated, possibly due to
+ faster access to memory since it's more likely to be in cache rather than main memory.
+
+ For medium-grained locking, it was clear that the correlation of the numbers had a big effect. As N grew larger, the
+  number of locks got smaller, and there became more contention over the lower number of locks. The correlated case
+  was more efficient because as N grew larger, it became more likely that the source and destination numbers were
+  going to be using the same lock, so there was less overall contention. Nevertheless, as N grew larger, it was
```

Comment:
great analysis of the tradeoff here. I observe the same trend except that my curve fluctuates a bit as N increases.